### PR TITLE
[CodeGen] Tail Duplication: Remove irrelevant phi node workaround

### PR DESCRIPTION
**Pull request by AZero13**
The comment is quite old, and removing it and then running the old test, it passed without asserting.

### CODE REVIEW

**Comment by AtariDreams** [comment]
@nikic @arsenm What do you think about this?

File: llvm/lib/CodeGen/TailDuplicator.cpp [hunk 1]
```diff
@@ -669,27 +669,6 @@ bool TailDuplicator::shouldTailDuplicate(bool IsSimple,
       return false;
   }
 
-  // Check if any of the successors of TailBB has a PHI node in which the
-  // value corresponding to TailBB uses a subregister.
-  // If a phi node uses a register paired with a subregister, the actual
-  // "value type" of the phi may differ from the type of the register without
-  // any subregisters. Due to a bug, tail duplication may add a new operand
-  // without a necessary subregister, producing an invalid code. This is
-  // demonstrated by test/CodeGen/Hexagon/tail-dup-subreg-abort.ll.
```
Comment (arsenm) [inline llvm/lib/CodeGen/TailDuplicator.cpp:678]: Can you remove the REQUIRES: asserts in this test, and add some generated checks? We could also write a targeted MIR test to make sure this works now

File: llvm/test/CodeGen/Hexagon/tail-dup-subreg-abort.ll [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 ; Tail duplication can ignore subregister information on PHI nodes, and as
 ; a result, generate COPY instructions between registers of different classes.
 ; This could lead to HexagonInstrInfo::copyPhysReg aborting on an unhandled
 ; src/dst combination.
 ;
 define i32 @foo(i32 %x, i64 %y) nounwind {
+; CHECK-LABEL: foo:
+; CHECK:       // %bb.0: // %entry
+; CHECK-NEXT:    {
+; CHECK-NEXT:     r1:0 = combine(r3,r2)
```
Comment (arsenm) [inline llvm/test/CodeGen/Hexagon/tail-dup-subreg-abort.ll:13]: This looks like too much was optimized out, that I doubt whatever MIR pattern was here didn't survive late enough

Comment (AtariDreams) [reply llvm/test/CodeGen/Hexagon/tail-dup-subreg-abort.ll:13]: Alright, so is there a way to see all the optimizations done

Comment (arsenm) [reply llvm/test/CodeGen/Hexagon/tail-dup-subreg-abort.ll:13]: -print-changed/-print-after-all?

**Pull request closed without merging**